src/tests/ftest/control/dmg_pool_query_test.py

```diff
@@ -0,0 +1,162 @@
+"""
+from __future__ import print_function
+
+from dmg_utils import pool_query, get_pool_query_info
+from ior_test_base import IorTestBase
+from test_utils_pool import TestPool
+
+
+class DmgPoolQueryTest(IorTestBase):
+    """Test Class Description:
+    Simple test to verify the pool query command of dmg tool.
+    :avocado: recursive
+    """
+    # pylint: disable=too-many-ancestors
+    def setUp(self):
+        "Set up for dmg pool query."
+        super(DmgPoolQueryTest, self).setUp()
+
+        # Init the pool
+        self.pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+        self.pool.get_params(self)
+        self.pool.create()
+
+        # Get the host list with port to provide the dmg command
+        self.port = self.params.get("port", "/run/server_config/*")
+        self.host_p = ["{}:{}".format(host, self.port)
+                       for host in self.hostlist_servers]
+
+    def test_pool_query_basic(self):
+        """
+        JIRA ID: DAOS-2976
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Provided a valid pool UUID, verify the output received from
+        pool query command.
+        :avocado: tags=all,tiny,pr,hw,dmg,pool_query,basic
+        """
+        self.log.info("Running dmg pool query")
+        dmg_out = pool_query(self.bin, self.host_p, self.pool.pool.uuid)
+
+        # Parse output
+        d_info = get_pool_query_info(dmg_out.stdout)
+        self.log.info("dmg values found: %s", d_info)
+
+        e_info = self.params.get("exp_vals", "/run/*")
+        self.log.info("Expected values are: %s", e_info)
+
+        # Verify
+        if d_info != e_info:
+            self.fail("dmg pool query expected output: {}".format(e_info))
+
+    def test_pool_query_inputs(self):
+        """
+        JIRA ID: DAOS-2976
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Verify the inputs that can be provided to 'query --pool'
+        argument of the dmg pool subcommand.
+        :avocado: tags=all,tiny,pr,hw,dmg,pool_query,basic
+        """
+        # Get test UUID
+        uuid = self.params.get("uuid", '/run/pool_uuids/*/')
+        self.log.info("Using test UUID: %s", uuid[0])
+
+        self.log.info("Running dmg pool query")
+        dmg_out = pool_query(self.bin, self.host_p, uuid[0])
+
+        # Verify
+        self.log.info("Test is expected to finish with: %s", uuid[1])
+        if dmg_out is None:
+            exception = 2
+        else:
+            if dmg_out.exit_status == 0:
+                exception = None
+            else:
+                exception = 1
+
+        if uuid[1] == "FAIL" and exception is None:
+            self.log.error("Command was expected to fail")
+            self.fail("Test failed, dmg pool query command output: {}".format(
+                dmg_out.stdout))
+        elif uuid[1] == "PASS" and exception is not None:
```

Comment:
It looks like there's no PASS case in the yaml. Do you need this block?

Reply:
I was debating on whether to put that block in the code in case future cases are added that do include the PASS case. To answer your question, it's not necessary at the moment.

Reply:
I see. Thanks for the clarification.